packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -73,8 +81,13 @@ class RouteSettings {
     this.isInitialRoute: false
   });
 
+  /// The name of the route (e.g., "/settings").
   final String name;
+
+  /// The set of keys that are most useful for constructoring [Hero] transitions.
   final Set<Key> mostValuableKeys;
+
+  /// Whether this route is the very first route being pushed onto this [Navigator].
```

Comment:
maybe add something about how this route won't be pushed off? I guess we expose this so you know not to show a back button, so maybe say that?